cli/deployments/eirini.go

```diff
@@ -0,0 +1,285 @@
+			return helpers.Kubectl("apply -f " + file)
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+	}
+
+	message = "Creating Eirini namespace for workloads"
+	out, err = helpers.SpinnerWaitCommand(message,
+		func() (string, error) {
+			file := path.Join(releaseDir, "deploy", "workloads", "namespace.yml")
+			return helpers.Kubectl("apply -f " + file)
+
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+	}
+
+	if err = k.patchNamespaceForQuarks(c, "eirini-workloads"); err != nil {
+		return err
+	}
+	if err = k.patchNamespaceForQuarks(c, "eirini-core"); err != nil {
+		return err
+	}
+
+	message = "Creating eirini certs using quarks-secret"
+	out, err = helpers.SpinnerWaitCommand(message,
+		func() (string, error) {
+			return helpers.KubectlApplyEmbeddedYaml(eiriniQuarksYaml)
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+	}
+
+	for _, component := range []string{"core", "events", "metrics", "workloads"} {
+		message := "Deploying eirini " + component
+		out, err := helpers.SpinnerWaitCommand(message,
+			func() (string, error) {
+				dir := path.Join(releaseDir, "deploy", component)
+				return helpers.Kubectl("apply -f " + dir)
+			},
+		)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+		}
+	}
+
+	domain, err := options.GetString("system_domain", eiriniDeploymentID)
+	if err != nil {
+		return err
+	}
+	// Create secrets and services accounts
+	if err = k.createClusterRegistryCredsSecret(c, false); err != nil {
+		return err
+	}
+	if err = k.createClusterRegistryCredsSecret(c, true); err != nil {
+		return err
+	}
+	if err = k.createGitCredsSecret(c, domain); err != nil {
+		return err
+	}
+	if err = k.patchServiceAccountWithSecretAccess(c, "eirini"); err != nil {
+		return err
+	}
+
+	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-api", 300); err != nil {
+		return errors.Wrap(err, "failed waiting Eirini api deployment to come up")
+	}
+	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-metrics", 300); err != nil {
+		return errors.Wrap(err, "failed waiting Eirini metrics deployment to come up")
+	}
+
+	emoji.Println(":heavy_check_mark: Eirini deployed")
+
+	return nil
+}
+
+func (k Eirini) GetVersion() string {
+	return eiriniVersion
+}
+
+func (k Eirini) Deploy(c kubernetes.Cluster, options kubernetes.InstallationOptions) error {
+
+	_, err := c.Kubectl.CoreV1().Namespaces().Get(
+		context.Background(),
+		eiriniDeploymentID,
+		metav1.GetOptions{},
+	)
+	if err == nil {
+		return errors.New("Namespace " + eiriniDeploymentID + " present already")
+	}
+
+	emoji.Println(":ship:Deploying Eirini")
+	err = k.apply(c, options)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k Eirini) Upgrade(c kubernetes.Cluster, options kubernetes.InstallationOptions) error {
```

Comment:
Dummy copy-paste code. Upgrading/Deleting/etc need more sophisticated handling. That's our of scope for this PR.